Dragonflight/AlgetharAcademy/EchoOfDoragosa.lua

```diff
@@ -71,10 +78,14 @@ function mod:AstralBreath(args)
 	self:CDBar(args.spellId, 32.8)
 end
 
-function mod:PowerVacuum(args)
-	self:Message(args.spellId, "red")
-	self:PlaySound(args.spellId, "alarm")
-	self:CDBar(args.spellId, 23.1)
+function mod:UNIT_SPELLCAST_SUCCEEDED(_, _, _, spellId)
+	if spellId == 388820 then -- Power Vacuum
+		powervacuumCount = powervacuumCount + 1
+		self:StopBar(388822)
+		self:Message(388822, "red")
+		self:PlaySound(388822, "alarm")
+		self:CDBar(388822, powervacuumCount == 1 and 24 or 29)
```

Comment:
This is a good find here, I timed this as well and found 24.2 and 29.1 if you wanted to increase this precision.

pull:23.1, 24.3, 29.1, 29.2, 29.1, 29.2, 29.1, 29.1, 29.2

Reply:
You could also change this to a :Bar instead of :CDBar since the timer seems well-understood now.

Reply:
> This is a good find here, I timed this as well and found 24.2 and 29.1 if you wanted to increase this precision.
> 
> pull:23.1, 24.3, 29.1, 29.2, 29.1, 29.2, 29.1, 29.1, 29.2

![](https://i.gyazo.com/8dcfa6f31d945559ab14cad4c3295054.png)

![](https://i.gyazo.com/7e25dc8bafca32eca8dd62264d0640e6.png)

your timers happen more often but I found lower timers aswell, so not sure which one to take.